quadriga/src/test/java/edu/asu/diging/quadriga/core/service/impl/AppManagerTest.java

```diff
@@ -0,0 +1,54 @@
+package edu.asu.diging.quadriga.core.service.impl;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+
+import edu.asu.diging.quadriga.core.data.AppRepository;
+import edu.asu.diging.quadriga.core.model.impl.App;
+import edu.asu.diging.quadriga.web.forms.AppForm;
+
+
+public class AppManagerTest {
+
+    
+    @Mock
+    private AppRepository appRepo;
+    
+    @InjectMocks
+    private AppManager managerToTest;
+    
+    private App app1;
+    private String ID1 = "ID1";
+   
+    
+    
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        app1 = new App();
+        app1.setId(ID1);
+        app1.setName("name");
+        app1.setDescription("testApp");
+        Mockito.when(appRepo.save(app1)).thenReturn(app1);
+        
+    }
+    
+    @Test
+    public void test_addApp_success() {
+        AppForm appForm = new AppForm();
+        String name = "name";
+        String desc = "testApp";
+        appForm.setName(name);
+        appForm.setDescription(desc);
+        App app = managerToTest.addApp(appForm);
+        Mockito.verify(appRepo).save(app);
+        Assert.assertEquals(name,app.getName());
+        Assert.assertEquals(desc,app.getDescription());
```

Comment:
you also want to test that the app has an id.

Reply:
Looking at your code, there is actually a potential bug that is related to this. Your addApp method should return the saved app it gets from the repo (instead of saving and returning the app that was passed to the repository), since only that one will have the new id, which the caller might need. If you do that, then you will need to mock the save method of the repository anyways and you can use that to set an id. Does this make sense?